.github/workflows/ci.yml

```diff
@@ -0,0 +1,28 @@
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```

Comment:
Does this mean that it runs this workflow on a PR to main and on MERGE to main? 
Is this a duplicate process then? Or are we just making sure that things are as they're meant to be?

Reply:
I think we need to do the check on the merged main branch also, because we can merge several pull requests after each other and end up with something new in main as long as it doesn't have merge conflicts...